staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -132,8 +139,15 @@ func diffMaps(original, modified map[string]interface{}, t reflect.Type, diffOpt
 	if t.Kind() == reflect.Ptr {
 		t = t.Elem()
 	}
+	// replaceKeysList will contains all the keys with non-nil value.
+	replaceKeysList := make([]interface{}, 0, len(modified))
 
 	for key, modifiedValue := range modified {
+		// Collect all keys with non-nil value.
+		if diffOptions.ReplaceKeys && modifiedValue != nil {
```

Comment:
// If we are building the retainkeys directive, include the keys for each element that is not empty